File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(*args):
    def _gcd(n, m):
        if m == 0:
            return n
        else:
            return _gcd(m, n % m)

    if len(args) < 2:
        return args[0]
    else:
        n = args[0]
        for m in args[1:]:
            n = _gcd(n, m)
        return n


def lcm(*args):
    def _lcm(n, m):
        return n * m // gcd(n, m)

    if len(args) < 2:
        return args[0]
    else:
        n = args[0]
        for m in args[1:]:
            n = _lcm(n, m)
        return n


n, m = map(int, input().split())

print(gcd(n, m))
print(lcm(n, m))

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import static java.lang.System.in;
import static java.lang.System.out;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int num1 = Integer.parseInt(st.nextToken());
        int num2 = Integer.parseInt(st.nextToken());

        if (num1 < num2) {
            int temp = num1;
            num1 = num2;
            num2 = temp;
        }

        bw.write(gcd(num1, num2) + "\n");
        bw.write(lcm(num1, num2) + "\n");

        bw.flush();
        br.close();
        bw.close();
    }

    private static int lcm(int num1, int num2) {
        return num1 * num2 / gcd(num1, num2);
    }

    private static int gcd(int num1, int num2) {
        if (num2 == 0) {
            return num1;
        }
        return gcd(num2, num1 % num2);
    }

}
